app/models/reports/result.rb

```diff
@@ -198,16 +201,31 @@ def registrations_for_percentage_calculation_with_ltfu(period)
       end
     end
 
-    def registrations_for_percentage_calculation(period, key)
-      if key == :controlled_patients_with_ltfu
-        return registrations_for_percentage_calculation_with_ltfu(period)
+    def registrations_for_percentage_calculation_absolute(period)
+      if quarterly_report?
+        self[:cumulative_registrations][period.previous] || 0
+      else
+        cumulative_registrations_for(period)
       end
+    end
 
+    def registrations_for_percentage_calculation_denominator(period)
```

Comment:
This name is a bit confusing - maybe `adjusted_registrations_for_percentage calculation` ?

Reply:
Yeah I wasn't sure about the naming here. I'm thinking of refactoring the percentage calculation so we don't have to do the `if else` based on the key at all.

Reply:
yeah these names are tough.  I think prepending the registration type (ie cumulative vs adjusted) could help if we have all these different methods going on.

I suppose another possible interface is:

```ruby
def registrations_for(period, type:)
  raise ArgumentError unless type.in?([:adjusted, :cumulative])
```

Something like that.  Not sure if that would end up better or worse.

Reply:
I've put the various denominators in a [hash for now](https://github.com/simpledotorg/simple-server/pull/2039/files#diff-2dd4f8a2ef5cd4e627113d439678eed40a6ad1adcc5d5de9f242dfe2e53459ddL198).